node/src/actors/session/handlers.rs

```diff
@@ -575,16 +589,75 @@ fn handshake_verack(session: &mut Session) {
     flags.verack_rx = true;
 }
 
+fn check_beacon_compatibility(
+    current_beacon: &LastBeacon,
+    received_beacon: &LastBeacon,
+) -> Result<(), HandshakeError> {
+    match current_beacon
+        .highest_block_checkpoint
+        .checkpoint
+        .cmp(&received_beacon.highest_block_checkpoint.checkpoint)
+    {
+        // current_checkpoint < received_checkpoint: received beacon is ahead of us
+        Ordering::Less => Ok(()),
```

Comment:
Can we also check that the provided epoch is not ahead of the epoch we truly should be at?

Reply:
Yes, but we need to make sure that we always set `highest_block_checkpoint` to the current epoch, except when the node is not synced, than the `highest_block_checkpoint` should be the epoch of the last block.

Reply:
Ok, this will make it impossible to bootstrap a testnet after the epoch 0. This is because new nodes always use (0, genesis_hash) as the beacon, so if the current epoch is not 0 other peers will reject them. And since all the nodes are in the same state, they will never be able to start to mine blocks.

Reply:
OK forget about that then!

Reply:
Oh right, I check that the epoch is equal, but instead we can check if the epoch is equal or lower. This will allow to bootstrap after epoch 0 because nodes will be able to peer with other nodes that are on a compatible beacon. But is that what we want?